packages/x-charts/src/internals/plugins/corePlugins/useChartInteractionListener/useChartInteractionListener.ts

```diff
@@ -0,0 +1,153 @@
+'use client';
+import * as React from 'react';
+import {
+  createUseGesture,
+  dragAction,
+  pinchAction,
+  wheelAction,
+  moveAction,
+  hoverAction,
+} from '@use-gesture/react';
+import { ChartPlugin } from '../../models';
+import {
+  UseChartInteractionListenerSignature,
+  AddInteractionListener,
+  AddMultipleInteractionListeners,
+  ChartInteraction,
+  ChartInteractionHandler,
+} from './useChartInteractionListener.types';
+
+type ListenerRef = Map<ChartInteraction, Set<ChartInteractionHandler<any, any>>>;
+
+const preventDefault = (event: Event) => event.preventDefault();
+
+// Create our own useGesture hook with the actions we need
+// Better for tree shaking
+const useGesture = createUseGesture([
+  dragAction,
+  pinchAction,
+  wheelAction,
+  moveAction,
+  hoverAction,
+]);
+
+export const useChartInteractionListener: ChartPlugin<UseChartInteractionListenerSignature> = ({
+  svgRef,
+}) => {
+  const listenersRef = React.useRef<ListenerRef>(new Map());
+
+  const retriggerEvent = React.useCallback((interaction: ChartInteraction, state: any) => {
+    const listeners = listenersRef.current.get(interaction);
+    const memo = !state.memo ? new Map<Function, any>() : state.memo;
+
+    if (listeners) {
+      listeners.forEach((callback) => {
+        const result = callback({ ...state, memo: memo.get(callback) });
+        if (result) {
+          memo.set(callback, result);
+        }
+      });
+    }
+
+    return memo;
+  }, []);
+
+  useGesture(
+    {
+      onDrag: (state) => retriggerEvent('drag', state),
+      onDragStart: (state) => retriggerEvent('dragStart', state),
+      onDragEnd: (state) => retriggerEvent('dragEnd', state),
+      onPinch: (state) => retriggerEvent('pinch', state),
+      onPinchStart: (state) => retriggerEvent('pinchStart', state),
+      onPinchEnd: (state) => retriggerEvent('pinchEnd', state),
+      onWheel: (state) => retriggerEvent('wheel', state),
+      onWheelStart: (state) => retriggerEvent('wheelStart', state),
+      onWheelEnd: (state) => retriggerEvent('wheelEnd', state),
+      onMove: (state) => retriggerEvent('move', state),
+      onMoveStart: (state) => retriggerEvent('moveStart', state),
+      onMoveEnd: (state) => retriggerEvent('moveEnd', state),
+      onHover: (state) => retriggerEvent('hover', state),
+    },
+    {
+      target: svgRef,
+      eventOptions: {
+        passive: false,
+      },
+      drag: {
+        pointer: {
+          // We can allow customizing the number of pointers
+          buttons: 1,
+          keys: false,
+        },
+        preventScroll: true,
```

Comment:
This enables the logic that prevents accidental scrolling on mobile. Pressing the touch for a moment will allow pan-y interacting in the zoom.